Zend/zend_attributes.c

```diff
@@ -95,6 +95,24 @@ static void validate_allow_dynamic_properties(
 	scope->ce_flags |= ZEND_ACC_ALLOW_DYNAMIC_PROPERTIES;
 }
 
+static void validate_attribute(
+	zend_attribute *attr, uint32_t target, zend_class_entry *scope)
+{
+	const char *msg = NULL;
+	if (scope->ce_flags & ZEND_ACC_TRAIT) {
+		msg = "Cannot apply #[Attribute] to trait %s";
```

Comment:
Please use the fully-qualified name in error messages for copy-and-paste safety for use in namespaces:
```suggestion
		msg = "Cannot apply #[\\Attribute] to trait %s";
```

Reply:
What is being copied - here the error is that you are using an attribute already where it isn't allowed
I just went with the existing pattern for validating `#[AllowDynamicProperties]`, but I can send a follow-up to change them both

Reply:
> Only classes can be marked with #[ZendTestAttribute]

That one should also be adjusted for completeness then.

Reply:
Added a leading `\`